macos/src/main/java/com/github/weisj/darklaf/platform/macos/JNIDecorationsMacOS.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Jannis Weis
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.github.weisj.darklaf.platform.macos;
+
+import com.github.weisj.darklaf.platform.NativeUtil;
+import com.github.weisj.darklaf.util.SystemInfo;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class JNIDecorationsMacOS {
+
+    private static final Logger LOGGER = Logger.getLogger(JNIDecorationsMacOS.class.getName());
+    private static boolean supported;
+    private static boolean loaded;
+
+    public static native void installDecorations(final long hwnd);
+
+    public static native void uninstallDecorations(final long hwnd);
+
+    public static native void setTitleEnabled(final long hwnd, final boolean enabled);
+
+    public static native void setDarkTheme(final long hwnd, final boolean darkEnabled);
+
+    public static native boolean isFullscreen(final long hwnd);
+
+    public static native double getTitleFontSize(final long hwnd);
+
+    /**
+     * Load the decorations-library if necessary.
+     *
+     * @return true if successful and library wasn't already loaded.
+     */
+    public static boolean updateLibrary() {
+        boolean oldLoaded = loaded;
+        if (!supported) {
+            supported = loadLibrary();
+        }
+        return supported && !oldLoaded;
```

Comment:
The logic here is not clear. Are the two flags really required?

Isn't `return !loaded && loadLibrary()` enough?

Reply:
This actually comes from an old version, where reloading a change in status should have an effect. Right now it only has to return whether the library is loaded or not. Will change later today.